app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -24,17 +28,51 @@ def get_middleware_servers
           @ems.eaps(feed).map do |eap|
             @eaps << eap
             server = parse_middleware_server(eap)
+
+            machine_id = @ems.machine_id(eap.feed)
+            host_instance = find_host_by_bios_uuid(machine_id)
+            if host_instance.nil?
+              host_instance = find_host_by_bios_uuid(alternate_machine_id(machine_id))
+            end
+
+            unless host_instance.nil?
+              server[:lives_on_id] = host_instance.try(:id)
+              server[:lives_on_type] = host_instance.try(:type)
+            end
+
             @data[:middleware_servers] << server
             @data_index.store_path(:middleware_servers, :by_nativeid, server[:nativeid], server)
           end
         end.flatten
       end
 
+      def alternate_machine_id(machine_id)
+        # see BZ #1294461 : here, we'll try to adjust the machine ID to the format from that bug
+        machine_id[6,2] + machine_id[4,2] + machine_id[2,2] + machine_id[0,2] + '-' +
+            machine_id[10,2] + machine_id[8,2] + '-' +
+            machine_id[14,2] + machine_id[12,2] + '-' +
+            machine_id[16,4] + '-' +
+            machine_id[20, 16]
+      end
+
+      def find_host_by_bios_uuid(machine_id)
+        identity_system = machine_id.downcase
+        Vm.find_by(:uid_ems => identity_system,
+                   :type    => uuid_provider_types) if identity_system
+      end
+
+      def uuid_provider_types
+        %w(ManageIQ::Providers::Redhat::InfraManager::Vm ManageIQ::Providers::Openstack::CloudManager::Vm ManageIQ::Providers::Vmware::InfraManager::Vm)
```

Comment:
I'll remove the ones we are not testing at the moment. This makes it easier to identify what is in scope for this PoC and what is not.